.github/workflows/test-java-nightly.yml

```diff
@@ -0,0 +1,42 @@
+name: Java SLT Nightly
+
+on:
+  schedule:
+    - cron: "30 11 * * *" # 11:30 AM UTC daily
+
+jobs:
+  java-tests:
+    name: Execute Java SLT
+    runs-on: [self-hosted, skylake40]
+    container:
+      image: ghcr.io/feldera/feldera-dev:f4797ad926773a38483275405728a9eb3b9005b5
+      options: --user=ubuntu
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # The docker container when executed in the action runs with a different home directory
+      # than we set in the dev container (?), hence this step is necessary (sigh)
+      # https://github.com/actions/runner/issues/863
+      - name: Rustup set default toolchain
+        run: rustup default stable
+
+      - name: Cache Rust dependencies
+        uses: Swatinem/rust-cache@v2
+
+      - name: Cache Maven dependencies
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2/repository
+          key: maven-${{ runner.os }}-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            maven-${{ runner.os }}-
+
+      # TODO: Avoid doing this twice and ideally download & reuse the JAR from build-java.yml workflow
+      - name: Run build.sh
+        run: ./build.sh
+        working-directory: ./sql-to-dbsp-compiler
+
+      - name: Run SLT tests
+        run: mvn test --no-transfer-progress -q -B -Dsurefire.failIfNoSpecifiedTests=false -Dtest=RotateTests
```

Comment:
are you sure this only runs SLT tests and not both?

Reply:
It only runs the tests in the class Rotate Tests. But it does build the compiler